### PR TITLE
Logzio exporter update (queue management) + Update documentation (new configuration options, updated example)

### PR DESCRIPTION
**Pull request by yotamloe**
**Description:** <Describe what has changed.>
Logz.io [exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/logzioexporter) update 🚀
The main changes are a new implementation of an in-memory queue to store traces, data compression with gzip, and queue configuration options.
**Link to tracking Issue:** <Issue number if applicable>
#3736 and #3702
**Testing:** <Describe what testing was performed and which tests were added.>
Added tests for new configuration options

From logzioexporter directory:
`go test -v -covermode=atomic -race`
output:
```
PASS
coverage: 96.6% of statements
```
**Documentation:** <Describe the documentation added.>
Added documantation for new configuration options + updated example

### CODE REVIEW

File: exporter/logzioexporter/README.md [hunk 1]
```diff
@@ -42,10 +48,10 @@ receivers:
...
     config:
       scrape_configs:
-      - job_name: 'ratelimiter'
-        scrape_interval: 15s
-        static_configs:
-        - targets: [ "0.0.0.0:8889" ]
+        - job_name: 'ratelimiter'
+          scrape_interval: 15s
+          static_configs:
+            - targets: [ "0.0.0.0:8889" ]
```
Comment (mx-psi) [inline exporter/logzioexporter/README.md:54]: Split to separate PR?

File: exporter/logzioexporter/go.mod [hunk 2]
```diff
@@ -4,16 +4,12 @@ go 1.16
...
-	github.com/mattn/go-colorable v0.1.7 // indirect
-	github.com/onsi/ginkgo v1.14.1 // indirect
-	github.com/onsi/gomega v1.10.2 // indirect
+	github.com/logzio/jaeger-logzio v1.0.3
 	github.com/stretchr/testify v1.7.0
 	go.opentelemetry.io/collector v0.31.0
 	go.opentelemetry.io/collector/model v0.31.0
 	go.uber.org/zap v1.18.1
 	google.golang.org/protobuf v1.27.1
-	gopkg.in/square/go-jose.v2 v2.5.1 // indirect
```
Comment (mx-psi) [inline exporter/logzioexporter/go.mod:18]: How are these changes related to queue management or solving the linked issue? If they are not please either document it on the PR description or if possible split into separate PR

File: exporter/logzioexporter/config.go [hunk 3]
```diff
@@ -27,6 +27,9 @@ type Config struct {
 	MetricsToken            string `mapstructure:"metrics_token"`   // Your Logz.io Metrics Token, can be found at https://docs.logz.io/user-guide/accounts/finding-your-metrics-account-token/
 	Region                  string `mapstructure:"region"`          // Your Logz.io 2-letter region code, can be found at https://docs.logz.io/user-guide/accounts/account-region.html#available-regions
 	CustomEndpoint          string `mapstructure:"custom_endpoint"` // Custom endpoint to ship traces to. Use only for dev and tests.
+	DrainInterval           int    `mapstructure:"drain_interval"`
+	QueueCapacity           int64  `mapstructure:"queue_capacity"`
+	QueueMaxLength          int    `mapstructure:"queue_max_length"`
```
Comment (mx-psi) [inline exporter/logzioexporter/config.go:32]: Can you document the settings here too with a comment as it is done with the others above?

File: exporter/logzioexporter/go.mod (hunk 2 above)
Comment (yotamloe) [reply exporter/logzioexporter/go.mod:18]: @mx-psi in this version of `jaeger-logzio` we implemented options to use in memory queue before that traces were stored on the fs which caused permission issue (#3736 and #3702) when users deploy the collector in a container because the default container user cannot write to the virtual fs ([PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2925/files))

Comment (yotamloe) [reply exporter/logzioexporter/go.mod:18]: these changes are already were merged in a previous PR (#4386) see [file](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/logzioexporter/go.mod)

File: exporter/logzioexporter/config.go (hunk 3 above)
Comment (yotamloe) [reply exporter/logzioexporter/config.go:32]: @mx-psi Thanks for your help. Sure.

File: exporter/logzioexporter/README.md (hunk 1 above)
Comment (yotamloe) [reply exporter/logzioexporter/README.md:54]: @mx-psi Help me to understand why, This part was in previous versions of this exporter.

File: exporter/logzioexporter/go.mod (hunk 2 above)
Comment (yotamloe) [reply exporter/logzioexporter/go.mod:18]: About queue management, the update in `jaeger-logzio` made it possible to add the new configuration options:(`DrainInterval`, `QueueCapacity `, `QueueMaxLength`)

Comment (mx-psi) [reply exporter/logzioexporter/go.mod:18]: I see, makes sense. Thanks for explaining

File: exporter/logzioexporter/README.md (hunk 1 above)
Comment (yotamloe) [reply exporter/logzioexporter/README.md:54]: You can check out the [file](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/logzioexporter/README.md)

Comment (mx-psi) [reply exporter/logzioexporter/README.md:54]: Maintainers on this repo like to split things into different PRs for better git history and easier review, so I was suggesting splitting this. I am fine with keeping it here if you document this change on the PR description too :)

Comment (yotamloe) [reply exporter/logzioexporter/README.md]: @mx-psi I understand. Thanks a lot for your help. I checked this specific change and fixed the indentation.

**Review by mx-psi — APPROVED** [review]
EasyCLA is having an incident (tracked on open-telemetry/community#795) so we may have to wait for it to be fixed, but otherwise this looks good to me :)
